packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -176,10 +207,32 @@ export class SummarizerNode implements IRootSummarizerNode {
 			}
 		}
 
-		// This should come from wipLocalPaths in normal cases, or from the latestSummary
-		// if parentIsFailure or parentIsReused is true.
-		// If there is no latestSummary, clearSummary and return before reaching this code.
-		assert(!!localPathsToUse, 0x1a5 /* "Tracked summary local paths not set" */);
+		/**
+		 * The absence of wip local path indicates that summarize was not called for this node. This can happen if:
+		 * 1. A child node was created after summarize was already called on the parent. For example, a data store
+		 * is realized (loaded) after summarize was called on it creating summarizer nodes for its DDSes. In this case,
+		 * parentSkipRecursion will be true and the if block above would handle it.
+		 * 2. A new node was created but summarize was never called on it. This can mean that the summary that is
+		 * generated may not have the data from this node. We should not continue, log and throw an error. This
+		 * will help us identify these cases and take appropriate action.
+		 */
+		if (localPathsToUse === undefined) {
+			const error = new LoggingError("NodeNotSummarized", {
```

Comment:
same question as before - can we somehow reuse code and reduce impact on bundle size?